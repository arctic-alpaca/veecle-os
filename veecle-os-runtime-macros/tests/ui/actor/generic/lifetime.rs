#[derive(Debug, PartialEq, Clone, Default, veecle_os_runtime::Storable)]
pub struct Sensor(pub u8);

#[veecle_os_runtime::actor]
async fn with_lifetime<'a>(
    _reader: veecle_os_runtime::Reader<'_, Sensor>,
    _writer: veecle_os_runtime::Writer<'_, Sensor>,
    #[init_context] _context: &'a (),
) -> core::convert::Infallible {
    core::future::pending().await
}

fn main() {
    let _ = veecle_os_runtime::execute! {
        store: [Sensor],
        actors: [WithLifetime<'static>: &()],
    };
}
