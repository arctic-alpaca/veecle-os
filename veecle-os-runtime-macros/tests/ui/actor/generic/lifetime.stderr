error: #[actor] functions cannot be generic over a lifetime
 --> tests/ui/actor/generic/lifetime.rs:5:24
  |
5 | async fn with_lifetime<'a>(
  |                        ^^

error[E0412]: cannot find type `WithLifetime` in this scope
  --> tests/ui/actor/generic/lifetime.rs:16:18
   |
16 |         actors: [WithLifetime<'static>: &()],
   |                  ^^^^^^^^^^^^ not found in this scope
